src/Compiler/Checking/NicePrint.fs

```diff
@@ -2058,6 +2059,7 @@ module TastDefinitionPrinting =
                 |> List.filter (fun (_, compgen, _) -> not compgen)
                 |> List.map p13
             else
+                // Always use the immediate (direct) interfaces
```

Comment:
You have added a comment, but not the implementation matching on `denv.showOnlyDirectBaseTypes`.
If it is set to false, existing behavior remains.
If set to true, only take the directly implemented interfaces.